kyuubi-server/src/main/scala/org/apache/kyuubi/session/SessionLimiter.scala

```diff
@@ -95,7 +95,10 @@ class SessionLimiterImpl(userLimit: Int, ipAddressLimit: Int, userIpAddressLimit
 
   private def decrLimitCount(key: String): Unit = {
     _counters.get(key) match {
-      case count: AtomicInteger => count.decrementAndGet()
+      case count: AtomicInteger =>
+        if (count.decrementAndGet() < 0) {
+          count.incrementAndGet()
+        }
```

Comment:
```suggestion
      case count: AtomicInteger if count.get > 0 =>
        val countValue = count.get
        count.compareAndSet(countValue, countValue - 1)
```
Back and forth operations on AtomicInteger will cause side effects, as no an extra thread-safe measure applied here. 
Consider using the CAS for changing the value of AtomicInteger. And even if the CAS failed, it will be ignored and it won't cause side effects or exceptions.

Reply:
@bowenliang123 hi, I have tried your suggestion, but tests failed.

Reply:
@pan3793 hi, my implementation can also pass tests, but @bowenliang123 proposed another implementation.Could you review it and give some suggestions? Thanks!

Reply:
A negative value is a situation that will happen but it's not acceptable for the counter.